app/src/main/java/org/mozilla/fenix/settings/PairFragment.kt

```diff
@@ -100,9 +110,40 @@ class PairFragment : Fragment(R.layout.fragment_pair), UserInteractionHandler {
                         it.onPermissionsResult(permissions, grantResults)
                     }
                 } else {
+                    cameraPermissionsDenied = true
                     findNavController().popBackStack(R.id.turnOnSyncFragment, false)
                 }
             }
         }
     }
+
+    private fun showPermissionsNeededDialog() {
+        AlertDialog.Builder(requireContext()).apply {
+            val spannableText = SpannableString(
+                resources.getString(R.string.camera_permissions_needed_message)
+            )
+            setMessage(spannableText)
+            setNegativeButton(R.string.camera_permissions_needed_negative_button_text) {
+                    dialog: DialogInterface, _ ->
+                dialog.cancel()
+            }
+            setPositiveButton(R.string.camera_permissions_needed_positive_button_text) {
+                    dialog: DialogInterface, _ ->
+                val intent: Intent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
Likewise here, why Version M?

Reply:
This specific intent, `Settings.ACTION_APPLICATION_DETAILS_SETTINGS`, was added in Android 6.0/Marshmallow/M. So for phones running older versions of android, we're just going to link to the help page in SUMO rather than supporting older intents

Reply:
Awesome, thank you for the comment